api/v1alpha1/datadogagent_default.go

```diff
@@ -615,12 +615,18 @@ func DefaultDatadogAgentSpecAgentApmUDS(apm *APMSpec) *APMUnixDomainSocketSpec {
 // DefaultDatadogAgentSpecAgentSystemProbe defaults the System Probe
 // This method can be re-run as part of the FeatureOverride
 func DefaultDatadogAgentSpecAgentSystemProbe(agent *DatadogAgentSpecAgentSpec) *SystemProbeSpec {
+	sysOverride := &SystemProbeSpec{}
+
 	if agent.SystemProbe == nil {
 		agent.SystemProbe = &SystemProbeSpec{Enabled: NewBoolPointer(defaultSystemProbeEnabled)}
-		return agent.SystemProbe
+		sysOverride = agent.SystemProbe
+	}
+
+	if agent.Security != nil && BoolValue(agent.Security.Runtime.Enabled) {
```

Comment:
Should we also check that `agent.Security.Runtime != nil` before `BoolValue(...)`?

Reply:
`agent.Security.Runtime` is not a pointer link: https://github.com/DataDog/datadog-operator/blob/main/api/v1alpha1/datadogagent_types.go#L648